pkg/cloudprovider/cloudprovider.go

```diff
@@ -87,6 +87,9 @@ func (c *CloudProvider) Create(ctx context.Context, machine *v1alpha5.Machine) (
 	if err != nil {
 		return nil, fmt.Errorf("resolving node template, %w", err)
 	}
+	machine.Annotations = lo.Assign(machine.Annotations, map[string]string{
```

Comment:
We should be adding tests around `cloudprovider_test.go` to ensure that the hash is actually returned on `Create`